sre-server/main.go

```diff
@@ -42,23 +103,218 @@ func run(args []string) error {
 		return err
 	}
 
+	// Create a shared informer factory
+	// TODO add defaultResync to the configuration
+	sharedInformers := informers.NewSharedInformerFactory(clientset, time.Minute*5)
+
+	// Create a deployment informer for all namespaces
+	deploymentInformer := sharedInformers.Apps().V1().Deployments().Informer()
+
+	// create a deployment lister for all namespaces
+	deploymentLister := sharedInformers.Apps().V1().Deployments().Lister()
+
+	// Start the informer factory to begin watching for changes
+	// we run the informer for the lifetime of the application
+	sharedInformers.Start(context.Background().Done())
+
+	// Wait for the cache to sync before exiting
+	if !cache.WaitForCacheSync(context.Background().Done(), deploymentInformer.HasSynced) {
+		log.Panicln("Timed out waiting for caches to sync")
+	}
+	log.Println("Informer is started")
+	// add router
+	router := mux.NewRouter()
 	// add handlers
-	http.Handle("/healthz", &healthzHandler{clientset: clientset})
+	router.Handle("/api/v1/namespaces/{namespace}/deployments/{name}/replicas",
+		&deploymentHandler{clientset: clientset, lister: deploymentLister})
+	router.Handle("/api/v1/deployments", &deploymentsHandler{lister: deploymentLister})
+	router.Handle("/api/v1/healthz", &healthzHandler{clientset: clientset, informer: deploymentInformer})
+	router.Handle("/api/v1/pingz", &pingzHandler{})
+
+	// Create HTTP Server with TLS Config
+	server := &http.Server{
```

Comment:
How do I upgrade the service without dropping requests?

Reply:
Will add graceful shutdown for it.